torch/onnx/verify.py

```diff
@@ -0,0 +1,262 @@
+        return np.array(elem, dtype=float)
+    elif isinstance(elem, dict):
+        flattened = []
+        for k in elem:
+            flattened += [_to_numpy(k)] + [_to_numpy(elem[k])]
+        return flattened
+    return elem
+
+
+def _convert_to_onnx(
+    model,
+    model_f: Union[str, io.BytesIO],
+    input=None,
+    opset_version=None,
+    do_constant_folding=True,
+    keep_initializers_as_inputs=True,
+    dynamic_axes=None,
+    input_names=None,
+    output_names=None,
+    fixed_batch_size=False,
+    training=None,
+    verbose=False,
+    ort_providers=_ORT_PROVIDERS,
+    ort_optim_on=True,
+):
+    input_copy = copy.deepcopy(input)
+
+    torch.onnx._export(
+        model,
+        input_copy,
+        model_f,
+        opset_version=opset_version,
+        do_constant_folding=do_constant_folding,
+        keep_initializers_as_inputs=keep_initializers_as_inputs,
+        dynamic_axes=dynamic_axes,
+        input_names=input_names,
+        output_names=output_names,
+        fixed_batch_size=fixed_batch_size,
+        training=training,
+        verbose=verbose,
+    )
+
+    try:
+        import onnxruntime  # type: ignore[import]
+    except ImportError:
+        raise RuntimeError("ONNXRuntime is required for export verification.")
+
+    # compute onnxruntime output prediction
+    so = onnxruntime.SessionOptions()
+    so.graph_optimization_level = (
+        onnxruntime.GraphOptimizationLevel.ORT_ENABLE_EXTENDED
+        if ort_optim_on
+        else onnxruntime.GraphOptimizationLevel.ORT_DISABLE_ALL
+    )
+    # suppress ort warnings.
+    # 0:Verbose, 1:Info, 2:Warning. 3:Error, 4:Fatal. Default is 2.
+    so.log_severity_level = 3
+    ort_sess = onnxruntime.InferenceSession(
+        model_f if isinstance(model_f, str) else model_f.getvalue(),
+        so,
+        providers=ort_providers,
+    )
+    return ort_sess
+
+
+def _inline_flatten_list(inputs, res_list):
+    for i in inputs:
+        res_list.append(i) if not isinstance(
+            i, (list, tuple)
+        ) else _inline_flatten_list(i, res_list)
+    return res_list
+
+
+def _unpack_to_numpy(values):
+    value_unpacked = []
+    for value in values:
+        value_unpacked.extend(unpack_quantized_tensor(value))
+    return [_to_numpy(v) for v in value_unpacked]
+
+
+def _run_ort(ort_sess, inputs):
+    kw_inputs = {}
+    if inputs and isinstance(inputs[-1], dict):
+        kw_inputs = inputs[-1]
+        inputs = inputs[:-1]
+    inputs = _unpack_to_numpy(_flatten_tuples(inputs))
+    ort_inputs = {}
+    for input_name, input in kw_inputs.items():
+        ort_inputs[input_name] = _to_numpy(input)
+    inputs = _to_numpy(inputs)
+    ort_sess_inputs = ort_sess.get_inputs()
+    for i, input in enumerate(inputs):
+        if i == len(ort_sess_inputs) or ort_sess_inputs[i].name in ort_inputs:
+            raise ValueError(
+                f"got too many positional inputs. inputs: {inputs}. kw_inputs: {kw_inputs}"
```

Comment:
I would add (who) got too many positional inputs for clarity.

Reply:
error will be thrown from _run_ort which should be straight forward, I guess it wouldn't hurt adding that though